aiida_vasp/io/parser.py

```diff
@@ -194,17 +230,76 @@ def get_quantity(self, quantity, settings, inputs=None):
                     if inputs[inp] is None and inp in self._parsable_items[quantity]['prerequisites']:
                         # The VaspParser was unable to provide the required input.
                         return {quantity: None}
-
             self._parsed_data = self._parse_file(inputs)
 
         return {quantity: self._parsed_data.get(quantity)}
 
+    def write(self, file_path):
+        """
+        Writes a VASP style file from the parsed Object.
+
+        For non input files this means simply copying the file.
+        """
+        if self._parsed_object is not None:
+            self._parsed_object.write(file_path)
+
+    @property
+    def _parsed_object(self):
+        """
+        Property to return the FileParsers _data_obj.
+
+        The data_obj is either an instance of one of the parsevasp parser classes,
+        which provide a write function, an instance of an aiida data node or an
+        instance of SingleFile in case that it is just a file and does not have
+        it's own 'write' method.
+
+        In particular FileParsers storing aiida data nodes. will have to override this.
+        """
+        return self._data_obj
+
+    @property
+    def data_obj(self):
```

Comment:
`_parsed_object` and `data_obj` properties are identical, except in name and that `_parsed_object` is protected. Is there a good reason not to merge the two?

Reply:
They are not always identical, only in the base class. The intention is that `_data_obj` will store the actual data the File Parser is operating on, either a file or an Aiida data node. Since the convention was that we store only either of those two, `_parsed_obj` was intended to return `ParserClass(_data_obj)`, which works for all `FileParsers` with corresponding parsevasp classes.  E.g. `_parsed_obj` would return `Poscar(_data_obj)` for the `PoscarParser`. The `_parsed_obj` is currently only used in `FileParser.write()` and will use the `ParserClass.write`. For all FileParsers without parsevasp class I put a dummy `ParserClass` called `SingleFile` which will provide a `write` method that in most cases just would copy the file. `_parsed_obj` was required previously because different FileParsers would parse their `_data_obj` differently. I am not sure, whether that is still the case.

Reply:
Not fixed. Martin, can you check this in detail?

Reply:
I can not remember anymore, why I added the public property `data_obj`. It is used only in `io.tests.test_poscar_potcar` and `io.tests.test_potcar_io`. It can probably be replaced by the protected version in both of those places and removed afterwards. Or replace every use `_data_obj` outside the FileParsers with the public version.

Reply:
Okey. We will leave this as is for now and polish at a later stage.